symbolication/datadog_uploader.go

```diff
@@ -16,18 +16,23 @@ import (
 	"runtime"
 	"time"
 
+	lru "github.com/elastic/go-freelru"
 	log "github.com/sirupsen/logrus"
 
 	"github.com/elastic/otel-profiling-agent/libpf"
 	"github.com/elastic/otel-profiling-agent/libpf/pfelf"
 	"github.com/elastic/otel-profiling-agent/libpf/vc"
 )
 
+const binaryCacheSize = 1000
+
 const sourceMapEndpoint = "/api/v2/srcmap"
 
 type DatadogUploader struct {
 	ddAPIKey  string
 	intakeURL string
+
+	uploadCache *lru.SyncedLRU[libpf.FileID, struct{}]
```

Comment:
the choice of file ID seems good
If we want to consider split debug, we would probably not cache the id of the split file, only of the main binary.

Reply:
Agreed 👍  That's how it's implemented in https://github.com/DataDog/otel-profiling-agent/pull/25 AFAICT 👌